namespaces/nsenter/nsenter.c

```diff
@@ -151,6 +151,15 @@ void nsenter()
 			exit(1);
 		}
 	}
+	// read continue signal from fd 3. the signal is 1
+	// this is so that the parent can put the proces inside correct cgroups
+	// before it has a chance to fork
+	unsigned char s;
+	if (read(3, &s, 1) != 1 || s != '1') {
```

Comment:
@crosbymichael i do a `read(2)` here to read 1 byte from the pipe ( which should block until it reads 1 byte iirc ), and write that byte in `execin.go`.

```
+	// finish the cgroup setup, signal to the cmd to continue
+	if _, err := parent.WriteString("1"); err != nil {
 		return terminate(err)
```